File: opcua-webservice-spring/examples/uawebservicetest/src/main/java/org/opcfoundation/uawebservicetest/testdigitaltwin/DigitalTwinSpaceTest.java
package org.opcfoundation.uawebservicetest.testdigitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.opcfoundation.webserver.digitaltwin.DigitalTwinSpace;

public class DigitalTwinSpaceTest extends DigitalTwinSpace {
    public static DigitalTwinDirectoryTestType digitalTwinDirectoryTestType;
    public static DigitalTwinTestType digitalTwinTestType;

    public static SubmodelTestType submodelTestType;
    public static DynamicSubmodelTestType dynamicSubmodelTestType;

    public static ElementCollectionTestAType elementCollectionTestAType;
    public static ElementCollectionTestBType elementCollectionTestBType;
    public static ReferenceElementTestType referenceElementTestType;
    public static EventElementTestType eventElementTestType;
    public static ElementListTestType elementListTestType;

    public static EnumTestDataType enumTestDataType;
    public static EventTestType eventTestType;

    public DigitalTwinSpaceTest()
    {
        super("opcfoundation/DigitalTwinSpaceTest");
    }

    @Override
    public void onStartUp()
    {
        System.out.println("DigitalTwinSpaceTest.onStartUp");

        enumTestDataType = new EnumTestDataType(this);
        eventTestType = new EventTestType(this);

        elementListTestType = new ElementListTestType(this);
        referenceElementTestType = new ReferenceElementTestType(this);

        eventElementTestType = new EventElementTestType(eventTestType,this);
        elementCollectionTestAType = new ElementCollectionTestAType(enumTestDataType,this);
        elementCollectionTestBType = new ElementCollectionTestBType(referenceElementTestType, elementListTestType, this);
        submodelTestType = new SubmodelTestType(elementCollectionTestAType, elementCollectionTestBType, referenceElementTestType, elementListTestType, eventElementTestType,this);
        dynamicSubmodelTestType = new DynamicSubmodelTestType(this);
        digitalTwinTestType = new DigitalTwinTestType(submodelTestType, dynamicSubmodelTestType, this);
        digitalTwinDirectoryTestType = new DigitalTwinDirectoryTestType(this);

        addRepository(digitalTwinDirectoryTestType, "Entry", new LocalizedText("入口"), new LocalizedText("测试用入口节点"));
    }
}

File: opcua-webservice-node/examples/testserver/testdigitaltwin/DigitalTwinSpaceTest.ts
import { UaLocalizedText } from "opcua-webapi-ts";
import { DigitalTwinSpace } from "../../../src/server/digitaltwin";
import { DigitalTwinDirectoryTestType } from "./DigitalTwinDirectoryTestType";
import { DigitalTwinTestType } from "./DigitalTwinTestType";
import { DynamicSubmodelTestType } from "./DynamicSubmodelTestType";
import { ElementCollectionTestAType } from "./ElementCollectionTestAType";
import { ElementCollectionTestBType } from "./ElementCollectionTestBType";
import { ElementListTestType } from "./ElementListTestType";
import { EnumTestDataType } from "./EnumTestDataType";
import { EventElementTestType } from "./EventElementTestType";
import { EventTestType } from "./EventTestType";
import { ReferenceElementTestType } from "./ReferenceElementTestType";
import { SubmodelTestType } from "./SubmodelTestType";

export class DigitalTwinSpaceTest extends DigitalTwinSpace {
    static digitalTwinDirectoryTestType: DigitalTwinDirectoryTestType;
    static digitalTwinTestType: DigitalTwinTestType;
    static submodelTestType: SubmodelTestType;
    static dynamicSubmodelTestType: DynamicSubmodelTestType;
    static elementCollectionTestAType: ElementCollectionTestAType;
    static elementCollectionTestBType: ElementCollectionTestBType;
    static referenceElementTestType: ReferenceElementTestType;
    static eventElementTestType: EventElementTestType;
    static elementListTestType: ElementListTestType;
    static enumTestDataType: EnumTestDataType;
    static eventTestType: EventTestType;

    constructor() {
        super("opcfoundation/DigitalTwinSpaceTest");
    }

    override async onStartUp(): Promise<void> {
        console.log("DigitalTwinSpaceTest.onStartUp");

        DigitalTwinSpaceTest.enumTestDataType = new EnumTestDataType(this);
        DigitalTwinSpaceTest.eventTestType = new EventTestType(this);

        DigitalTwinSpaceTest.elementListTestType = new ElementListTestType(this);
        DigitalTwinSpaceTest.referenceElementTestType = new ReferenceElementTestType(this);

        DigitalTwinSpaceTest.eventElementTestType = new EventElementTestType(DigitalTwinSpaceTest.eventTestType, this);
        DigitalTwinSpaceTest.elementCollectionTestAType = new ElementCollectionTestAType(DigitalTwinSpaceTest.enumTestDataType, this);
        DigitalTwinSpaceTest.elementCollectionTestBType = new ElementCollectionTestBType(
            DigitalTwinSpaceTest.referenceElementTestType,
            DigitalTwinSpaceTest.elementListTestType,
            this,
        );
        DigitalTwinSpaceTest.submodelTestType = new SubmodelTestType(
            DigitalTwinSpaceTest.elementCollectionTestAType,
            DigitalTwinSpaceTest.elementCollectionTestBType,
            DigitalTwinSpaceTest.referenceElementTestType,
            DigitalTwinSpaceTest.elementListTestType,
            DigitalTwinSpaceTest.eventElementTestType,
            this,
        );
        DigitalTwinSpaceTest.dynamicSubmodelTestType = new DynamicSubmodelTestType(this);
        DigitalTwinSpaceTest.digitalTwinTestType = new DigitalTwinTestType(
            DigitalTwinSpaceTest.submodelTestType,
            DigitalTwinSpaceTest.dynamicSubmodelTestType,
            this,
        );
        DigitalTwinSpaceTest.digitalTwinDirectoryTestType = new DigitalTwinDirectoryTestType(this);

        this.addRepository(
            DigitalTwinSpaceTest.digitalTwinDirectoryTestType,
            "Entry",
            new UaLocalizedText("入口"),
            new UaLocalizedText("测试用入口节点"),
        );
    }
}